.github/workflows/sanity-test.yaml

```diff
@@ -0,0 +1,314 @@
+          fi
+
+      - name: Repo init test
+        id: repo-init
+        env:
+          TEST_RESULT: "test_results"
+        run: |
+          set -euo pipefail 
+          prefix=`date +"%Y-%m-%d-%T"`
+
+          ./corso repo init s3 \
+          --hide-progress \
+          --prefix $prefix \
+          --bucket  ${CORSO_BUCKET}  2>&1 | tee $TEST_RESULT/initrepo.txt 
+
+          if ! grep -q 'Initialized a S3 repository within bucket' $TEST_RESULT/initrepo.txt 
+          then
+              echo "repo could not be initiated"
+              exit 1
+          fi
+          
+          echo result="$prefix" >> $GITHUB_OUTPUT 
+
+      # run the tests
+      - name: Repo connect test
+        run: |
+          set -euo pipefail
+          ./corso repo connect s3 \
+          --hide-progress \
+          --prefix ${{ steps.repo-init.outputs.result }} \
+          --bucket  ${CORSO_BUCKET} 2>&1 | tee $TEST_RESULT/connect.txt 
+
+          if ! grep -q 'Connected to S3 bucket' $TEST_RESULT/connect.txt 
+          then
+              echo "repo could not be connected"
+              exit 1
+          fi
+
+      # run the tests
+      - name: Backup exchange test
+        id: exchange-test
+        run: |
+          ./corso backup create exchange \
+          --user "${CORSO_M365_TEST_USER_ID}" \
+           --hide-progress 2>&1 | tee $TEST_RESULT/backup_exchange.txt 
+
+          if ! grep -q 'Completed (0 errors)' $TEST_RESULT/backup_exchange.txt
+          then
+              echo "backup was not successfull"
+              exit 1
+          fi
+          echo result=$(grep -i -e 'Completed (0 errors)' $TEST_RESULT/backup_exchange.txt  | awk '{print $2}') >> $GITHUB_OUTPUT
+
+      # list the backup exhange
+      - name: Backup exchange list test
+        run: |
+          set -euo pipefail
+          ./corso backup list exchange \
+          --hide-progress 2>&1 | tee $TEST_RESULT/backup_exchange_list.txt
+
+          if ! grep -q  ${{ steps.exchange-test.outputs.result }} $TEST_RESULT/backup_exchange_list.txt 
+          then
+              echo "listing of backup was not successfull"
+              exit 1
+          fi
+
+      # AWS creds
+      - name: Configure AWS credentials from Test account
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          role-to-assume: ${{ secrets.AWS_IAM_ROLE }}
+          role-session-name: integration-testing
+          aws-region: us-east-1
+
+      # test exchange restore
+      - name: Backup exchange restore
+        id: exchange-restore-test
+        run: |
+          set -euo pipefail
+          ./corso restore exchange \
+          --hide-progress \
+          --backup "${{ steps.exchange-test.outputs.result }}" 2>&1 | tee $TEST_RESULT/exchange-restore-test.txt
+          echo result=$(grep -i -e 'Restoring to folder ' $TEST_RESULT/exchange-restore-test.txt | sed "s/Restoring to folder//" )  >> $GITHUB_OUTPUT
+        
+      - name: Restoration check
```

Comment:
```
 Restore folder:  Corso_Restore_10-Mar-2023_05:40:02
Folder name:  Archive count:  0
Folder name:  Conversation History count:  0
Folder name:  Corso_Restore_10-Mar-2023_05:20:35 count:  0
Folder name:  Corso_Restore_10-Mar-2023_05:24:55 count:  0
Folder name:  Corso_Restore_10-Mar-2023_05:27:38 count:  0
Folder name:  Corso_Restore_10-Mar-2023_05:33:04 count:  0
Folder name:  Corso_Restore_10-Mar-2023_05:43:22 count:  0
Folder name:  Corso_Restore_10-Mar-2023_05-10-21.365951 count:  0
Folder name:  Corso_Restore_10-Mar-2023_05-26-31.405257 count:  0
```

The restored list of items seem to be `0` 🤔

Reply:
Added that for testing. Have removed it!